libs/bluetooth/bluetooth_utils.c

```diff
@@ -210,6 +210,105 @@ const char *bleVarToUUIDAndUnLock(ble_uuid_t *uuid, JsVar *v) {
+      if (privacyModeVar && jsvIsString(privacyModeVar)) {
+        if (jsvIsStringEqual(privacyModeVar, "off")) {
+          privacy->privacy_mode = BLE_GAP_PRIVACY_MODE_OFF;
+        } else if (jsvIsStringEqual(privacyModeVar, "device_privacy")) {
+          privacy->privacy_mode = BLE_GAP_PRIVACY_MODE_DEVICE_PRIVACY;
+        } else {
+          invalidOption = true;
+        }
+      } else {
+        invalidOption = true;
+      }
+      jsvUnLock(privacyModeVar);
+    }
+    // private addr type
+    {
+      JsVar *privacyAddrTypeVar = jsvObjectGetChildIfExists(options, "private_addr_type");
+      if (privacyAddrTypeVar && jsvIsString(privacyAddrTypeVar)) {
+        if (jsvIsStringEqual(privacyAddrTypeVar, "random_private_resolvable")) {
+          privacy->private_addr_type = BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_RESOLVABLE;
+        } else if (jsvIsStringEqual(privacyAddrTypeVar, "random_private_non_resolvable")) {
+          privacy->private_addr_type = BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_NON_RESOLVABLE;
+        } else {
+          invalidOption = true;
+        }
+      } else {
+        invalidOption = true;
+      }
+      jsvUnLock(privacyAddrTypeVar);
+    }
+    // private addr cycle s
+    {
+      JsVar *privateAddrCycleSVar = jsvObjectGetChildIfExists(options, "private_addr_cycle_s");
+      if (privateAddrCycleSVar && jsvIsInt(privateAddrCycleSVar)) {
+        privacy->private_addr_cycle_s = jsvGetInteger(privateAddrCycleSVar);
+      } else {
+        invalidOption = true;
+      }
+      jsvUnLock(privateAddrCycleSVar);
+    }
+    return !invalidOption;
+  }
+  return false;
+}
+
+JsVar *blePrivacyToVar(pm_privacy_params_t *privacy) {
+  JsVar *result = jsvNewObject();
+  if (!result) return 0;
+  if (privacy) {
+    char *privacy_mode_str = "";
+    switch (privacy->privacy_mode) {
+      case BLE_GAP_PRIVACY_MODE_OFF:
+        privacy_mode_str = "off";
+        break;
+      case BLE_GAP_PRIVACY_MODE_DEVICE_PRIVACY:
+        privacy_mode_str = "device_privacy";
+        break;
+    }
+    char *addr_type_str = "";
+    switch (privacy->private_addr_type) {
+      case BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_RESOLVABLE:
+        addr_type_str = "random_private_resolvable";
+        break;
+      case BLE_GAP_ADDR_TYPE_RANDOM_PRIVATE_NON_RESOLVABLE:
+        addr_type_str = "random_private_non_resolvable";
+        break;
+    }
+    jsvObjectSetChildAndUnLock(result, "privacy_mode", jsvNewFromString(privacy_mode_str));
+    jsvObjectSetChildAndUnLock(result, "private_addr_type", jsvNewFromString(addr_type_str));
+    jsvObjectSetChildAndUnLock(result, "private_addr_cycle_s", jsvNewFromInteger(privacy->private_addr_cycle_s));
+    return result;
+  }
+  jsvUnLock(result);
+  return 0;
+}
+
+bool blePrivacyVarEqual(JsVar *a, JsVar *b) {
+  return (jsvIsUndefined(a) && jsvIsUndefined(b)) ||
+         (jsvIsObject(a) && jsvIsObject(b) &&
+          jsvIsEqual(jsvObjectGetChildIfExists(a, "privacy_mode"),
```

Comment:
These will all leak as you're not unlocking the result of jsvObjectGetChildIfExists

Reply:
Thanks for pointing that out!  
Should be fixed now.